auth/credentials/oauth_client_test.go

```diff
@@ -0,0 +1,182 @@
+package credentials
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// TokenResponse represents a mock OAuth token response.
+type TokenResponse struct {
+	AccessToken  string `json:"access_token"`
+	TokenType    string `json:"token_type"`
+	ExpiresIn    int    `json:"expires_in"`
+}
+
+// MockOAuthTokenEndpoint creates a mock OAuth token endpoint that returns a fake token.
+func MockOAuthTokenEndpoint(token string) *httptest.Server {
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Ensure the request uses POST and contains proper grant_type
+		if r.Method != http.MethodPost || r.FormValue("grant_type") != "client_credentials" {
+			http.Error(w, "invalid request", http.StatusBadRequest)
+			return
+		}
+
+		// Create mock token response
+		tokenResponse := TokenResponse{
+			AccessToken:  token,
+			TokenType:    "Bearer",
+			ExpiresIn:    3600,
+		}
+
+		// Encode and return the JSON response
+		w.Header().Set("Content-Type", "application/json")
+		_ = json.NewEncoder(w).Encode(tokenResponse)
```

Comment:
maybe do a require.NotError or something similar to check err?

Reply:
💪 Can do! This code is mocked part of the test. We can print error etc.